python/main.py

```diff
@@ -64,24 +75,53 @@ def hello():
 class AddItemResponse(BaseModel):
     message: str
 
+class Item(BaseModel):
+    name: str
+    category: str
+    image_name: str
+
+def insert_item(item: Item):
+    # STEP 4-1: add an implementation to store an item
+    try:
+        with open("items.json", "r") as f:
+            data = json.load(f)
+    except (FileNotFoundError, json.JSONDecodeError):
+        data = {"items": []}
+
+        # Append new item
+    data["items"].append({"name": item.name, "category": item.category, "image_name": item.image_name})
+
+        # Write back to the file
+    with open("items.json", "w", encoding="utf-8") as f:
+        json.dump(data, f, indent=2, ensure_ascii=False)
 
 # add_item is a handler to add a new item for POST /items .
-@app.post("/items", response_model=AddItemResponse)
-def add_item(
+@app.post("/items")
+async def add_item(
     name: str = Form(...),
     category: str = Form(...),
+    image: Optional[UploadFile] = File(None),
     db: sqlite3.Connection = Depends(get_db),
-):
-    if not name:
+):    
+    image_name = ""  # Default to empty string if no image is uploaded
+    if image is not None:
+        file_bytes = await image.read()
+        image_hash = hashlib.sha256(file_bytes).hexdigest()
+        image_name = f"{image_hash}.jpg"
+        image_path = images_dir / image_name
+        with open(image_path, "wb") as f:
+            f.write(file_bytes)#
+
+    if not name or not category:
```

Comment:
You shouldn't include `not category` condition.
```suggestion
    if not name:
```